app/controllers/reset_pins_controller.rb

```diff
@@ -6,9 +6,38 @@ class ResetPinsController < ApplicationController
 
   def index; end
 
+  def change_form; end
+
+  def change
+    @response = symphony_client.change_pin(*change_pin_with_token_params)
+    case @response.status
+    when 200
+      flash[:success] = t 'mylibrary.change_pin.success_html'
+      redirect_to login_path
+    else
+      flash[:error] = t 'mylibrary.change_pin.error_html'
+      redirect_to reset_pin_path
+    end
+  end
+
   def reset
-    @response = symphony_client.reset_pin(params['library_id'])
+    @response = symphony_client.reset_pin(
+      params['library_id'],
+      change_pin_with_token_unencoded_url
+    )
     flash[:success] = t 'mylibrary.reset_pin.success_html', library_id: params['library_id']
-    redirect_to login_path
+    render action: :index
+  end
+
+  private
+
+  def change_pin_with_token_params
+    params.require(%I[token pin])
+  end
+
+  ##
+  # Hacky workaround Rails routing helpers to have the <> not encoded
+  def change_pin_with_token_unencoded_url
+    change_pin_with_token_url('foo').gsub('foo', '<RESET_PIN_TOKEN>')
```

Comment:
Where is `RESET_PIN_TOKEN` coming from? Symphony land?

Reply:
Yep. Symphony looks for the string `<RESET_PIN_TOKEN>` and injects the actual token in a URL crafted in the email sent to the user.

Reply:
Got it; would you mind documenting that in the method comment?

Reply:
Sure thing.. added.